src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -119,18 +119,33 @@ int TPCHandler::opensocket_callback(void *clientp,
                                     curlsocktype purpose,
                                     struct curl_sockaddr *aInfo)
 {
-// See what kind of address will be used to connect
-//
-if (purpose == CURLSOCKTYPE_IPCXN && clientp)
-   {XrdNetAddr thePeer(&(aInfo->addr));
-    ((TPCLogRecord *)clientp)->isIPv6 =  (thePeer.isIPType(XrdNetAddrInfo::IPv6)
-                                      && !thePeer.isMapped());
-   }
+  //Return a socket file descriptor (note the clo_exec flag will be set).
+  int fd = XrdSysFD_Socket(aInfo->family, aInfo->socktype, aInfo->protocol);
```

Comment:
It seems to me that there's no point in continuing if you get `fd < 0` here, so maybe add an if statement like
```cpp
if (fd < 0)
    return CURL_SOCKET_BAD;
```
and below just `return fd;`

Reply:
Yes correct, done :)